tests/test_interface_service.rb

```diff
@@ -0,0 +1,239 @@
+
+    # Reset feature to clean up switch
+    config('no feature vni', 'feature vni')
+
+    # Create a global encap config
+    config('encapsulation profile vni vni_500_5000', 'dot1q 500  vni 5000',
+           'encapsulation profile vni vni_600_6000', 'dot1q 600  vni 6000',
+           'encapsulation profile vni vni_700_7000', 'dot1q 700  vni 7000',
+           'encapsulation profile vni vni_800_8000', 'dot1q 800  vni 8000')
+
+    # Test interface name
+    intf = 'ethernet9/1'
+    config("default int #{intf}")
+    intf
+  end
+
+  def test_create_destroy
+    intf = setup_encapsulation_profile_vni
+
+    # TEST Create / Destroy and svc_vni_ids hash builder
+    i5 = InterfaceService.new(intf, 5)
+    assert_equal(1, InterfaceService.svc_vni_ids[intf].count)
+    i6 = InterfaceService.new(intf, 6)
+    i7 = InterfaceService.new(intf, 7)
+    assert_equal(3, InterfaceService.svc_vni_ids[intf].count)
+    i6.destroy
+    assert_equal(2, InterfaceService.svc_vni_ids[intf].count)
+    i5.destroy
+    i7.destroy
+  end
+
+  def test_shutdown
+    intf = setup_encapsulation_profile_vni
+
+    i5 = InterfaceService.new(intf, 5)
+    # Test shutdown
+    i5.shutdown = false
+    refute(i5.shutdown)
+    i5.shutdown = true
+    assert(i5.shutdown)
+    i5.shutdown = false
+    refute(i5.shutdown)
+    i5.shutdown = i5.default_shutdown
+    assert(i5.shutdown)
+  end
+
+  def test_encapsulation_profile_vni
+    intf = setup_encapsulation_profile_vni
+
+    i5 = InterfaceService.new(intf, 5)
+
+    # Test removal when profile not present
+    i5.encapsulation_profile_vni = ''
+    assert_empty(i5.encapsulation_profile_vni)
+
+    # Add one
+    i5.encapsulation_profile_vni = 'vni_500_5000'
+    assert_equal('vni_500_5000', i5.encapsulation_profile_vni)
+
+    # Change it
+    i5.encapsulation_profile_vni = 'vni_700_7000'
+    assert_equal('vni_700_7000', i5.encapsulation_profile_vni)
+
+    # Test default when profile present
+    i5.encapsulation_profile_vni = 'vni_700_7000'
+    i5.encapsulation_profile_vni = i5.default_encapsulation_profile_vni
+    assert_equal(i5.default_encapsulation_profile_vni,
+                 i5.encapsulation_profile_vni)
+  end
+
+  # interface vlan_mapping is not technically part of the interface service
+  # but it shares most of the same dependencies so it is tested here instead
+  # of test_interface.rb
+  def setup_vlan_mapping
+    # This property has several dependencies:
+    #  - VDC support
+    #  - Specific linecard (F3)
+    #  - Bridge Domain Configuration
+    #  - Feature vni
+
+    # Check for supported platform
+    skip("Test not supported on #{node.product_id}") if
+      cmd_ref.lookup('vdc', 'all_vdcs').config_get_token.nil?
+
+    # This test requires a specific linecard; as such we will hard-code the
+    # module location and skip the test if not found.
+    # Example 'show mod' output to match against:
+    # '9    12     10/40 Gbps Ethernet Module          N7K-F312FQ-25      ok'
+    slot = 9
```

Comment:
It'll work in any slot. I like your idea to figure it out based on the show command. I'll fix post-merge.